packages/overlay/src/integrations/sentry/components/insights/agents/AITraceFlow.tsx

```diff
@@ -0,0 +1,196 @@
+        aiTrace.operation.toLowerCase().includes(searchLower) ||
+        aiTrace.toolCalls.some(tool => tool.toolName.toLowerCase().includes(searchLower)),
+    );
+  }, [aiTraces, inputValue]);
+
+  if (aiTraces.length === 0) {
+    return (
+      <div className="p-6 text-center">
+        <p className="text-primary-300">No AI interactions found in this trace</p>
+      </div>
+    );
+  }
+
+  return (
+    <>
+      <div className="mx-6 mb-4 mt-2 flex gap-2">
+        <div className="bg-primary-950 text-primary-50 border-primary-600 hover:border-primary-500 relative flex h-auto w-full flex-1 gap-2 rounded-md border py-1 pl-4 pr-6 outline-none transition-all">
+          <input
+            className="text-primary-50 h-auto w-full flex-1 bg-transparent outline-none transition-all"
+            onChange={handleChange}
+            value={inputValue}
+            placeholder="Search AI interactions..."
+          />
+          {showReset && (
+            <CrossIcon
+              onClick={handleReset}
+              className="fill-primary-50 absolute right-1 top-[5px] cursor-pointer"
+              height={20}
+              width={20}
+            />
+          )}
+        </div>
+      </div>
+
+      <div className="min-h-0 flex-1 overflow-y-auto overflow-x-hidden px-6 pb-6 ">
+        <div className="space-y-4">
+          {filteredAITraces.map((aiTrace, index) => (
+            <AIFlowItem
+              key={aiTrace.id}
+              aiTrace={aiTrace}
+              isSelected={spanId === aiTrace.id}
+              traceId={traceId}
+              index={index}
+              total={filteredAITraces.length}
+            />
+          ))}
+        </div>
+      </div>
+    </>
+  );
+}
+
+function AITraceFlowContent({ traceId }: AITraceFlowProps) {
+  const getTraceById = useSentryStore(state => state.getTraceById);
+
+  const trace = getTraceById(traceId)!;
+  const startTimestamp = trace.start_timestamp;
+  const totalDuration = trace.timestamp - startTimestamp;
+
+  // extract AI traces from the trace
+  const aiTraces = useMemo(() => {
+    if (!trace.spanTree) return [];
+
+    const aiRootSpans = extractAllAIRootSpans(trace.spanTree);
+    return aiRootSpans
+      .map(({ span }) => createAITraceFromSpan(span))
+      .filter((aiTrace): aiTrace is SpotlightAITrace => aiTrace !== null)
+      .sort((a, b) => a.timestamp - b.timestamp); // Sort by timestamp
+  }, [trace]);
+
+  return (
+    <>
+      <div className="px-6 py-4">
+        <div className="text-primary-300 flex flex-1 items-center gap-x-1">
+          <div className="text-primary-200">
+            <DateTime date={trace.start_timestamp} />
+          </div>
+          <span>&mdash;</span>
+          <span>
+            <strong className="text-primary-200 font-bold">{getFormattedDuration(totalDuration)}</strong> with{" "}
+            <strong className="text-primary-200 font-bold">
+              {aiTraces.length} AI interaction{aiTraces.length !== 1 ? "s" : ""}
```

Comment:
I'd vary the whole string as in case we do translations in the future, it would make it much easier to do it.